platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -27,6 +58,61 @@ - (instancetype)initWithCoordinates:(const CLLocationCoordinate2D *)coords count
     return self;
 }
 
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+    self = [super initWithCoder:decoder];
+    if (self)
+    {
+        NSArray *coordinates = [decoder decodeObjectOfClass:[NSArray class] forKey:@"coordinates"];
+        CLLocationCoordinate2D *coords = (CLLocationCoordinate2D *)malloc([coordinates count] * sizeof(CLLocationCoordinate2D));
+        for (NSUInteger i = 0; i < [coordinates count]; i++)
+        {
+            coords[i] = CLLocationCoordinate2DMake([coordinates[i][0] doubleValue], [coordinates[i][1] doubleValue]);
+        }
+        _coordinates = { coords, coords + [coordinates count] };
+        free(coords);
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder
+{
+    [super encodeWithCoder:coder];
+    
+    NSMutableArray *coordinates = [NSMutableArray array];
+    for (auto coord : _coordinates) {
+        [coordinates addObject:@[@(coord.latitude), @(coord.longitude)]];
```

Comment:
`@encode` CLLocationCoordinate2D would work as long as we don't use it in combination with NSKeyed(Un)Archiver. It only gets the layout of the struct but no information about the name of the fields which is causing the archiver to raise an exception when trying to archive structs. Will make it unambiguous by representing it as a dictionary instead.